go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -165,59 +170,154 @@ func (oc *Controller) updateEgressFirewallWithRetry(egressfirewall *egressfirewa
 	})
 }
 
-func (ef *egressFirewall) addACLToJoinSwitch(joinSwitches []string, hashedAddressSetNameIPv4, hashedAddressSetNameIPv6 string) error {
+func (oc *Controller) addACLToJoinSwitch(joinSwitches []string, hashedAddressSetNameIPv4, hashedAddressSetNameIPv6 string, ef *egressFirewall) error {
 	for _, rule := range ef.egressRules {
-		var match string
 		var action string
+		var matchTargets []matchTarget
 		if rule.access == egressfirewallapi.EgressFirewallRuleAllow {
 			action = "allow"
 		} else {
 			action = "drop"
 		}
+		if rule.to.cidrSelector != "" {
```

Comment:
TODO (not necessarily this PR): move all validation to NewEgressFirewall (or somewhere similar)

Reply:
this isn't validation at this point, this is just a switch to determine if we are using a cidrSelector or dnsName because generating the match is different.

Reply:
Except the part where you parse the cidr :-)

Reply:
you are right I removed that

Reply:
you are right I removed that